### DISCUSSION TITLE
Setup Wizard

### DISCUSSION THREAD

**Discussion by leon-richardt** [Chatterino/chatterino2#4273] [Ideas]
# Introduction
Discoverability of features and the design of settings has been a long-standing pain point of Chatterino (cf. #1956 and links therein). Currently, Chatterino sets up reasonable default values for every setting, for some definition of "reasonable". However, what is considered reasonable by some may be considered unintuitive for others and lead to frustration when the expected behavior is not the default but buried deep inside the Settings page. Even worse, the setting one would need to change may only be described by confusing language inaccessible to non-expert users of the software.

I am advocating for a setup wizard that guides users through first installation and helps them set up important settings. (For some definition of "important" ...) That way, we would help make clear which settings we consider important instead of letting users wade through the Settings page on their own. In the wizard, we could use more space to explain what effect a setting has and provide justification for what we consider a reasonable default.

# Important Settings
What is considered an important setting may certainly be up to debate. Nonetheless, I believe there is some set of settings most contributors can agree on. This Discussion can serve as a place to suggest and discuss settings for inclusion in the setup wizard. Let me start with an initial list, sorted by sections in the Settings page.

- **Messages**:
    - "Hide deleted messages"
    - "Draw a line below the most recent message before switching applications." and its style settings
- **Emotes**:
    - "Show unlisted 7TV emotes"
    - "Emoji style"
    - Toggles for BTTV/FFZ/7TV emotes
- **Streamer Mode**:
    - All settings
- **Link Previews**:
    - All settings
- **Miscellaneous**:
    - "Show moderation messages"
    - "Show deletions of single messages"
    - "Only search for emote autocompletion at the start of emote names"
    - "Only search for username autocompletion with an @"
    - "Show Twitch whispers inline" and "Highlight received inline whispers"
    - "Load message history on connect" and related settings
    - "Strip leading mentions in replies"

**Comment by Mm2PL** [comment]
I'd like for a possible setup wizard to ask for login, as it seems like the first thing one should do with a chat app. Forcing the user to login is a bad decision. We need to find a balance, probably only suggesting login.

Here is another idea, what if we had some kind of presets for defaults like: streamer - disable "Show unlisted 7TV emotes", set streamer mode to auto, and so on.

**Comment by Mm2PL** [comment]
Maybe we should also consider adding options to just pick an image uploader

**Reply by leon-richardt** [reply]
I agree, could be an especially nice opportunity to provide "pre-sets" as also given on the [wiki page](https://wiki.chatterino.com/Image%20Uploader/). However, I recall some hesitance regarding setting i.nuuls.com as the default uploader because of the perhaps unexpected increase in users.

**Reply by Mm2PL** [reply]
What if the default was just "Disabled" and the user could just pick from the ones on the wiki or a custom one

**Reply by pajlada** [reply]
Default disabled, but pre-sets are fine. Maybe reach out to small hosters to see if it's fine first? @nuuls

**Reply by nuuls** [reply]
Yeah as long as I'm not the only option that's fine 👍
